### PR TITLE
Improve dex docs as per customer feedback

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/giantswarm/issues/31208

This PR:
- Adds section to map user and group to role
- Athena API server port 6443

### Things to check/remember before submitting

- If you made content changes

    - Run `make dev` to render and proofread content changes locally.
    - Bump `last_review_date` in the front matter header if you reviewed the entire page.

### CODE REVIEW

File: src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md [hunk 1]
```diff
@@ -444,6 +444,44 @@ If both `Dex` and `Athena` are configured correctly and you have installed `kube
 kubectl gs login https://api.test.example.io
 ```
 
+## Mapping Users and Groups to Roles
+
+The following YAML examples demonstrate how you can map a users or groups to roles in the cluster for access.
```
Comment (anvddriesch) [inline src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md:449]:
```suggestion
The following YAML examples demonstrate how you can map users or groups to roles in the cluster for access.
```

File: src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md [hunk 2]
```diff
@@ -444,6 +444,44 @@ If both `Dex` and `Athena` are configured correctly and you have installed `kube
...
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-admin
+subjects:
+- kind: User
+  name: you@example.io
+  apiGroup: rbac.authorization.k8s.io
+```
+
+This example grants cluster adin privileges to the user "you@example.io".
```
Comment (anvddriesch) [inline src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md:466]:
```suggestion
This example grants cluster admin privileges to the user "you@example.io".
```

**Review by anvddriesch — APPROVED** [review]
looks good to me but it might be good to have a link to general rbac documentation here as well.
We could also recommend using the `rbac-bootstrap-app` here since it simplifies this even more.

**Comment by tuladhar** [comment]
> looks good to me but it might be good to have a link to general rbac documentation here as well. We could also recommend using the `rbac-bootstrap-app` here since it simplifies this even more.

Yeah, good point.

**Comment by tuladhar** [comment]
Please re-review @anvddriesch

**Review by anvddriesch — APPROVED** [review]

File: src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md [hunk 3]
```diff
@@ -401,7 +401,7 @@ kubernetes:
     M...=
     -----END CERTIFICATE-----
   api:
-    address: https://api.test.example.io
+    address: https://api.test.example.io:6443
```
Comment (AndiDog) [inline src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md:404]: With CAPA, we default to Kubernetes API port 443 by now (changed from 6443)

**Review by ssyno — APPROVED** [review]

File: src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md (hunk 3 above)
Comment (tuladhar) [reply src/content/vintage/advanced/access-management/configure-dex-in-your-cluster/index.md:404]: Thanks, added note to use 443 for CAPA.

**Pull request merged**